### PR TITLE
Homework 1

### PR DESCRIPTION
**Pull request by amypetris**

### CODE REVIEW

**Comment by timand32** [comment]
0. It looks like you imported the first observation as a header in your DataFrame:
Screenshot from 2021-09-16 21-23-16
* Passing `header=None` to `read_csv()` is the easier way I know of having pandas ignore headers, but there's probably others.
* When you rename columns on line 18, you may want to add an extra dict entry for the class feature, (else the last column wouldn't have a descriptive name).
1. The plots look nice.  Adding transparency on the histogram might be something to consider, as it's hard to compare the 3 classes around 5cm-5.5cm with opaque colors.
2. Wrapping lines 86 - 89 into a function may save you from having to copy and paste printing code after every model.

File: HW1.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.tree import DecisionTreeClassifier
+
+
+def main():
+
+    # Load Data
+    iris_data = pd.read_csv("iris.data")
```
Comment (timand32) [inline HW1.py:15]: The first observation is being imported as a header, you may want to fix this.

File: HW1.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    # Rename columns
+    iris_data = iris_data.rename(
+        columns={
+            "5.1": "sepal_length",
+            "3.5": "sepal_width",
+            "1.4": "petal_length",
+            "0.2": "petal_width",
+        }
```
Comment (timand32) [inline HW1.py:24]: You may want to add an extra entry. in this dict.  Otherwise, the last column wouldn't have a descriptive name.

File: HW1.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    )
+
+    # Summary statistics
+    summary = pd.DataFrame.describe(iris_data)
+    print(summary)
+
+    # Plot and display figures
+    fig_hist = px.histogram(iris_data, x="sepal_length", color="Iris-setosa")
```
Comment (timand32) [inline HW1.py:32]: Transparency would make this plot a bit more clear.  I don't know if this is easy to do with plotly.express, however...

File: HW1.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            ("RandomForest", RandomForestClassifier(random_state=1234)),
+        ]
+    )
+
+    pipeline_RF.fit(X_orig, y)
+
+    probability = pipeline_RF.predict_proba(X_orig)
+    prediction = pipeline_RF.predict(X_orig)
+    print(f"Probability Random Forest: {probability}")
+    print(f"Predictions Random Forest: {prediction}")
```
Comment (timand32) [inline HW1.py:88]: You could refactor this into a function, so you wouldn't have to copy print code after each model you add.

**Review by timand32 — COMMENTED** [review]
Added my comments from earlier here.

File: HW1.py (hunk 1 above)
Comment (amypetris) [reply HW1.py]: Fixed this, added header=None

Reference: ae50f75027bf6f9296028511081cfe43ebea0912

File: HW1.py (hunk 2 above)
Comment (amypetris) [reply HW1.py]: Fixed.

Reference: ae50f75027bf6f9296028511081cfe43ebea0912

File: HW1.py [hunk 5]
```diff
@@ -3,90 +3,75 @@
...
-    print(summary)
 
-    # Plot and display figures
-    fig_hist = px.histogram(iris_data, x="sepal_length", color="Iris-setosa")
+def summary(df):
+    return df.describe()
+
+
+def plot(df):
+    fig_hist = px.histogram(df, x="sepal_length", color="class", opacity=0.5)
```
Comment (timand32) [inline HW1.py:32]: `opacity=` appears to control transparency for the entire graph, not individual classes.

Based on [this documentation](https://plotly.com/python/histograms/#overlaid-histogram), you may want to switch this `barmode="overlay"`

**Review by timand32 — COMMENTED** [review]
Ran `HW1.py` again, the data frame issues seem to be fixed.  The only suggestion I still have is that `opacity=` might not be the parameter you want for use your histogram (See my comment below).

File: HW1.py (hunk 5 above)
Comment (amypetris) [reply HW1.py]: Great suggestion! Fixed.

Reference: 74608ada84ac5b1ba2c16a8efccceaab33ff438b

**Review by timand32 — APPROVED** [review]
Ran it again, everything looks good.  I have no more suggestions I can think of!

File: HW1.py (hunk 1 above)
Comment (dafrenchyman) [reply HW1.py]: Nice catch!

**Pull request merged**